code/datums/gamemode/role/time_agent.dm

```diff
@@ -163,3 +235,42 @@
 	new /obj/item/weapon/grenade/chronogrenade/future(src)
 	new /obj/item/weapon/grenade/smokebomb(src)
 	new /obj/item/weapon/grenade/empgrenade(src)
+
+/obj/item/device/timeline_eraser
+	name = "timeline eraser"
+	desc = "Erase someone from the timeline. It has an unusual affinity against time travellers..."
+	icon_state = "jump_charge"
+	w_class = W_CLASS_SMALL
+	var/channeling = FALSE
+
+
+
+/obj/item/device/timeline_eraser/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	var/duration = 100
+	// TODO: Make the timestop, properly stop when the process is done
+	if(istype(target, /mob))
+		var/mob/M = target
+		if(istimeagent(M))
+			duration = 0
+
+	if(proximity_flag && !(target.flags & TIMELESS))
+		channeling = TRUE
+		spawn()
+			while(channeling)
+				//make the process obvious, but don't impede people interfering with it too much
+				timestop(src, 30, 7, 0, /mob/living/carbon/)
+				sleep(60)
+		if(do_after(user, target, duration))
+			delete_from_timeline(target, user)
+		channeling = FALSE
+	if(target.flags & TIMELESS)
+		to_chat(user, "<span class = 'warning'>The target is currently immune to temporal meddling.</span>")
+
+/obj/item/device/timeline_eraser/proc/delete_from_timeline(atom/target, mob/user)
+	if(istimeagent(user))
+		var/datum/role/R = user.mind.GetRole(TIMEAGENT)
+		if(R)
+			var/datum/objective/target/assassinate/erase/E = locate() in R.objectives.GetObjectives()
+			if(E)
+				E.check(target)
+	qdel(target)
```

Comment:
probably should also remove them from records at a minimum, but it would be great if you could take them out of PDA logs and tcomm server logs too

Reply:
records being sec/med? do wizards and similar get put in those automatically, or are you talking about if someone manually adds the antag to them?

Reply:
1. Yes
2. They do not get them automatically but yes if someone manually adds them. 
3. It doesn't appear(?) that you check to see if the eraser target is a time agent, so they could be erasing normal crew, no?

Reply:
eraser is intended to be used on normal crew in addition to other time agents, yeah; I was thinking too much about the twin erasing the original. I'll work on the crew's records later, sure.